### PR TITLE
tests: samples: remove duplicate filtering

### LINKED ISSUE #615: testcase filtering overload
We have many testcases doing filtering both on the architecture level and the platform level, which is redundant. Also many testcases are running the same test twice on the same SoC for no good reason, cleanup the tests and cleanup the filtering.

### PR DESCRIPTION
**Pull request by nashif**
This fixes #615.

### CODE REVIEW

**Review by tyrantlizard77 — COMMENTED** [review]
Any reason why we removed Arduino 101 from a bunch of these tests?

File: samples/net/zoap_server/sample.yaml [hunk 1]
```diff
@@ -5,7 +5,6 @@ tests:
 -   test:
         arch_whitelist: x86
         build_only: true
-        platform_exclude: quark_d2000_crb
```
Comment (galak) [inline samples/net/zoap_server/sample.yaml]: do you really want to remove this?

File: tests/drivers/dma/chan_blen_transfer/testcase.yaml [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard  quark_d2000_crb
         tags: drivers
 -   test_dma_shell:
-        arch_whitelist: x86
         build_only: true
         extra_args: CONF_FILE=prj_shell.conf
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/dma/chan_blen_transfer/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/dma/chan_blen_transfer/testcase.yaml [hunk 3]
```diff
@@ -1,13 +1,9 @@
 tests:
 -   test_dma:
-        arch_whitelist: x86
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard  quark_d2000_crb
```
Comment (galak) [inline tests/drivers/dma/chan_blen_transfer/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/gpio/gpio_basic_api/testcase.yaml [hunk 4]
```diff
@@ -1,7 +1,5 @@
 tests:
 -   test_gpio:
-        arch_whitelist: x86 arc
         build_only: true
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/gpio/gpio_basic_api/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/i2c/i2c_api/testcase.yaml [hunk 5]
```diff
@@ -1,11 +1,7 @@
 tests:
 -   test_i2c_x86:
-        arch_whitelist: x86
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/i2c/i2c_api/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/pinmux/pinmux_basic_api/testcase.yaml [hunk 6]
```diff
@@ -1,6 +1,4 @@
 tests:
 -   test_pinmux:
-        arch_whitelist: x86 arc
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/pinmux/pinmux_basic_api/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/rtc/rtc_basic_api/testcase.yaml [hunk 7]
```diff
@@ -1,6 +1,4 @@
 tests:
 -   test_rtc:
-        arch_whitelist: x86 arc
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/rtc/rtc_basic_api/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/spi/spi_basic_api/testcase.yaml [hunk 8]
```diff
@@ -1,7 +1,4 @@
 tests:
 -   test_spi:
-        arch_whitelist: x86 arc arm
-        platform_exclude: arduino_101_sss
-        platform_whitelist: quark_se_c1000_devboard
-            quark_d2000_crb arduino_101 galileo frdm_k64f
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb galileo frdm_k64f
```
Comment (galak) [inline tests/drivers/spi/spi_basic_api/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/uart/uart_basic_api/testcase.yaml [hunk 9]
```diff
@@ -1,13 +1,9 @@
 tests:
 -   test_uart:
-        arch_whitelist: x86
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/uart/uart_basic_api/testcase.yaml]: any reason arduino_101 was dropped?

File: tests/drivers/uart/uart_basic_api/testcase.yaml [hunk 10]
```diff
@@ -1,13 +1,9 @@
...
         tags: drivers
 -   test_uart_shell:
-        arch_whitelist: x86
         build_only: true
         extra_args: CONF_FILE=prj_shell.conf
-        platform_whitelist: quark_se_c1000_devboard
-            arduino_101 quark_d2000_crb
+        platform_whitelist: quark_se_c1000_devboard quark_d2000_crb
```
Comment (galak) [inline tests/drivers/uart/uart_basic_api/testcase.yaml]: any reason arduino_101 was dropped?

**Review by galak — CHANGES_REQUESTED** [review]
Mostly some questions, seems like you dropped arduino_101 from a number of places, maybe good to comment in the commit message on this.

Also, add some comment in the commit message about filter in both arch & platform...

File: samples/net/zoap_server/sample.yaml (hunk 1 above)
Comment (nashif) [inline samples/net/zoap_server/sample.yaml]: yes, we do not build net samples on this board by default.

File: tests/drivers/uart/uart_basic_api/testcase.yaml (hunk 10 above)
Comment (nashif) [inline tests/drivers/uart/uart_basic_api/testcase.yaml]: the arduino 101 is covered by quark_se_c1000_devboard for this particular driver test, the QA folks were using the whitelist for their own accounting. Ultimate goal is to remove this type of whitelisting and and provide more coverage by matching features between board and what the testscase covers.

File: samples/net/zoap_server/sample.yaml (hunk 1 above)
Comment (galak) [inline samples/net/zoap_server/sample.yaml]: if we don't build the samples, don't we still need the exclude than?

File: tests/drivers/uart/uart_basic_api/testcase.yaml (hunk 10 above)
Comment (galak) [inline tests/drivers/uart/uart_basic_api/testcase.yaml]: ok, that makes sense

File: samples/net/zoap_server/sample.yaml (hunk 1 above)
Comment (nashif) [inline samples/net/zoap_server/sample.yaml]: see 

https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/x86/quark_d2000_crb/quark_d2000_crb.yaml#L11

Comment (galak) [inline samples/net/zoap_server/sample.yaml]: ah

**Review by galak — APPROVED** [review]

**Pull request merged**